jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpChannelState.java

```diff
@@ -1073,6 +1073,22 @@ public Session getSession(boolean create)
             return null;
         }
 
+        @Override
+        public int hashCode()
+        {
+            // Override the implementation from the base class,
```

Comment:
No, the base class is `Attributes.Lazy`, which implements them based on the attributes map -- so that 2 different requests both with no attributes would be equal.

Reply:
Ah right, this is for the internal `ChannelRequest` class, not the outer `HttpChannelState` one.